### PR TITLE
Exception Parser

### PR DESCRIPTION
**Pull request by poulad**
This PR holds changes to exception parser API and feature goes along with changes in [Telegram.Bot.Extensions.Exceptions](https://github.com/TelegramBots/Telegram.Bot.Extensions.Exceptions) pacakge

### CODE REVIEW

File: src/Telegram.Bot/Exceptions/DefaultExceptionParser.cs [hunk 1]
```diff
@@ -93,5 +97,33 @@ private static string TryTruncateErrorDescription(string message, string descrip
...
+
+        /// <inheritdoc cref="IExceptionParser"/>
+        public async Task<Exception> ParseResponseAsync(HttpResponseMessage httpResponse)
+        {
+            Exception exception;
+
+            string responseJson = await httpResponse.Content.ReadAsStringAsync()
+                .ConfigureAwait(false);
+
+            var apiResponse = JsonConvert.DeserializeObject<JObject>(responseJson);
```
Comment (karb0f0s) [inline src/Telegram.Bot/Exceptions/DefaultExceptionParser.cs]: Do I need to deserialize response in each custom Exception Handler? How is that better then processing already deserialized data in current implementation? At lest this gives an overhead of having dependency on `Newtonsoft.Json` in extension.

File: src/Telegram.Bot/Exceptions/DefaultExceptionParser.cs [hunk 2]
```diff
@@ -93,5 +97,33 @@ private static string TryTruncateErrorDescription(string message, string descrip
...
+            else if (ok == false)
+            {
+                exception = new ApiRequestException(description);
+            }
+            else
+            {
+                exception = new ApiRequestException("Invalid status code: " + httpResponse.StatusCode);
+            }
+
+            return exception;
```
Comment (karb0f0s) [inline src/Telegram.Bot/Exceptions/DefaultExceptionParser.cs]: Do I understand correctly, base parser will only return ApiRequestException? And it seems that `HttpStatusCode` was lost on the way :(

**Comment by pouladpld** [comment]
@karb0f0s This is just PoC code and not real implementation. Feel free to change it.

**Comment by tuscen** [comment]
Superseded by [Telegram.Bot.Extensions.Exceptions](TelegramBots/Telegram.Bot.Extensions.Exceptions) package.

**Pull request closed without merging**